### PR TITLE
Do not override system properties for Windows OS. Fixes #9867

### LINKED ISSUE #9867: systemProperties is overriden in Windows OS for ApplicationContextCommand tasks
When calling the method configurePathingJar from the GrailsGradlePlugin class, it overrides the systemProperties set on the task previously in the ApplicationContextCommandTask. 

This is happen only for Windows OS and it seems it is related to the changes introduced in the PR #9728 by @Schlogen

### PR DESCRIPTION
**Pull request by inicolas**
@Schlogen could you please review this PR ?

### CODE REVIEW

File: grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy [hunk 1]
```diff
@@ -438,25 +438,35 @@ class GrailsGradlePlugin extends GroovyPlugin {
...
+                        }
                     }
                 }
 
-                tasks.withType(ApplicationContextCommandTask) { ApplicationContextCommandTask task ->
+                project.tasks.withType(ApplicationContextCommandTask) { ApplicationContextCommandTask task ->
                     task.dependsOn(pathingJarCommand)
-                    task.systemProperties = ['grails.env': 'development']
+                    task.doFirst {
+                        classpath = pathingClasspathCommand
```
Comment (Schlogen) [inline grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy]: Why is setting the classpath necessary for `pathingJarCommand`?

File: grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy [hunk 2]
```diff
@@ -438,25 +438,35 @@ class GrailsGradlePlugin extends GroovyPlugin {
...
-            Jar pathingJarCommand = createPathingJarTask(project, "pathingJarCommand", runtime, console)
+            final pathingJar = createPathingJarTask(project, "pathingJar", runtime)
+            final pathingClasspath = project.files("${project.buildDir}/classes/main",
+              "${project.buildDir}/resources/main", "${project.projectDir}/gsp-classes", pathingJar.archivePath)
+            final pathingJarCommand = createPathingJarTask(project, "pathingJarCommand", runtime, console)
+            final pathingClasspathCommand = project.files("${project.buildDir}/classes/main",
+              "${project.buildDir}/resources/main", "${project.projectDir}/gsp-classes", pathingJarCommand.archivePath)
 
-            GrailsExtension grailsExt = project.extensions.getByType(GrailsExtension)
+            final grailsExt = project.extensions.getByType(GrailsExtension)
```
Comment (Schlogen) [inline grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy]: Add types back for everything

Comment (inicolas) [inline grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy]: done

File: grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy (hunk 1 above)
Comment (inicolas) [inline grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy]: I did not pay attention that ApplicationContextCommandTask was a subclass of JavaExec.

**Pull request merged**